libraries/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -655,6 +660,12 @@ bool IotMqtt_IsSubscribed( IotMqttConnection_t mqttConnection,
                            IotMqttSubscription_t * const pCurrentSubscription )
 {
     bool status = false;
+    
+    if( pTopicFilter == NULL )
+    {
+    	return status;
+    }
```

Comment:
If we don't check whether `pTopicFilter` is NULL or not, we might call `_topicMatch` with `topicMatchParams.pTopicName`. Then, the `pTopicName` will be passed as an input for `strncmp` function. @gordonwang0